### ISSUE TITLE
Wrapping my App with <KeyboardProvider> adds space on top and bottom of the screen.

### ISSUE THREAD

**Issue by e-biba-ofti** [kirillzyusko/react-native-keyboard-controller#587]
**Describe the bug**
When wrapping my app with `KeyboardProvider` it adds space looks like it tries to do the same `<SafeAreaView>` does but I am already using `<SafeAreaView>` partially in my app as in some screens is not needed. If I just disable it `<KeyboardProvider enabled={false}>` then the app shows in full view.

**Expected behavior**
The app should show in full screen

**Screenshots**
![image](https://github.com/user-attachments/assets/38548942-6744-4f12-ab71-63094c44e2d8)

**Smartphone (please complete the following information):**

- Desktop OS: [e.g. Windows 10, MacOS 10.15.5]
- Device: Android
- OS: Android
- RN version: 0.73.2
- RN architecture: [e.g. old/new or paper/fabric]
- JS engine: [e.g. JSC, Hermes, v8]
- Library version: 1.13.3

**Comment by kirillzyusko** [comment]
@e-biba-ofti have you tried to specify [statusBarTranslucent](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/keyboard-provider#statusbartranslucent-) and [navigationBarTranslucent](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/api/keyboard-provider#navigationbartranslucent-) props on `KeyboardProvider` level?

**Comment by edisoni1995** [comment]
@kirillzyusko `navigationBarTransparent` seems to work fine but the `statusBarTranslucent` doesn't work. After checking it a bit more when I remove my status bar it works as expected or when I set my status bar as translucent it's working. But I don't want my status bar to be `translucent` as the content will be drawn under it.

**Comment by kirillzyusko** [comment]
@edisoni1995 when you wrap app in `KeyboardProvider` it will move to `edge-to-edge` mode and app will be drawn under status bar. Since `edge-to-edge` is not a default mode for RN app the `KeyboardProvider` will try to match default RN behavior and add paddings.

On your screenshot these white paddings most likely are caused by this fact (i. e. it's paddings from `KeyboardProvider` and for me it looks like you already handle safe-area paddings somewhere in your code). To remove these padding you will need to add `navigationBarTranslucent` and `statusBarTranslucent` props.

`statusBarTranslucent` doesn't work in your case because you are using `StatusBar` component and `translucent` is `false` by default, so you overwrite an initial config of `KeyboardProvider` and you are adding white padding again.

I think in your case it'll be safe to add `translucent` property to StatusBar - in this case it shouldn't draw anything under `StatusBar` because `react-native-keyboard-controller` uses [its own implementation](https://github.com/kirillzyusko/react-native-keyboard-controller/blob/main/android/src/main/java/com/reactnativekeyboardcontroller/modules/StatusBarManagerCompatModuleImpl.kt) (this implementation is compatible with `edge-to-edge` mode, unlike RN component) of `StatusBar` and it becomes to be used automatically when you use `KeyboardProvider`.

Can you share here two screenshots side by side:
- one how your current app screen looks right now without `react-native-keyboard-controller` integration;
- the second one is with `KeyboardProvider` (with translucent props) and with `<StatusBar translucent />` in place?

In my understanding these screenshots should be identical.

**Comment by e-biba-ofti** [comment]
Here you have it, I made a red border just to be able to tell that some part of the content is going under the status bar.
The image on the right is the one with `<StatusBar translucent />` and with translucent props for `KeyboardProvider`. 


![image](https://github.com/user-attachments/assets/2f9a8caf-4ccc-497e-a6cd-4681a4757d26) ![image](https://github.com/user-attachments/assets/edb042d6-5dc9-44ce-bc75-f02596e22348)

**Comment by kirillzyusko** [comment]
@e-biba-ofti would you be able to provide a minimal reproduction example?

Something is definetely going wrong here, but without a code I can not say what exactly 😔 

Would you be able also to try to comment out this line:

https://github.com/kirillzyusko/react-native-keyboard-controller/blob/7c9c450984161dcb76858184fcb2d249b4dcf8f4/android/src/main/java/com/reactnativekeyboardcontroller/modules/StatusBarManagerCompatModuleImpl.kt#L58

And re-run app again?

**Comment by e-biba-ofti** [comment]
@kirillzyusko that would be very difficult to create for now but I will try to find a minimal reproducable example. I did the change in the android file and re-build android but nothing changed.

**Comment by kirillzyusko** [comment]
@e-biba-ofti got you! Then minimal reproduction example is highly appreciate - it can significantly reduce the time needed for fixing this problem! ❤️

**Comment by RayKay91** [comment]
The issue will be resolved soon - it's been merged to `react-native-screens` but is yet unreleased. I posted this in another issue:

https://github.com/kirillzyusko/react-native-keyboard-controller/issues/592#issuecomment-2370922343

**Comment by e-biba-ofti** [comment]
Yes I am using `navigationBarOptions` and `react-native-screens@3.29.0`

**Comment by kirillzyusko** [comment]
Well @e-biba-ofti `react-native-screens@3.29.0` is highly unlikely to break something 🤔 

I really can't help here 😔 If you can provide a reproduction example that would be a dope and would significantly reduce time for fixing 🙏

**Comment by rranjan14** [comment]
I am also facing this issue on android. I am getting a white space at the top of the screen. The space is roughly equal to the height of the statusbar. This is happening with `react-native-screens >=3.31.0`

**Comment by kirillzyusko** [comment]
@rranjan14 what is the exact version of `react-native-screens` in `package-lock.json` or `yarn.lock`? Can you share a snippet of data from your `.lock` file?

**Comment by rranjan14** [comment]
<img width="1136" alt="Screenshot 2024-10-22 at 14 20 36" src="https://github.com/user-attachments/assets/3c499c41-da1e-494e-8253-940e4974884f">

here you go @kirillzyusko

**Comment by kirillzyusko** [comment]
@rranjan14 okay, you'll need to add a patch like https://github.com/kirillzyusko/react-native-keyboard-controller/blob/main/example/patches/react-native-screens%2B3.34.0.patch or apply changes from this PR: https://github.com/software-mansion/react-native-screens/pull/2301

Both patches are identical. Hopefully new version of `react-native-screens` will be released and that issue will go away 😅 

Alternatively you can downgrade your version to `3.31.0` (without `>=` modifiers).

**Comment by rranjan14** [comment]
@kirillzyusko I have been following this issue for some time now. I have tried this but to no avail sadly 😕

**Comment by kirillzyusko** [comment]
>  I did. And also, I am using SystemBars from react-native-edge-to-edge and not StatusBar from react-native or expo-status-bar

Well, interesting 🤔 Can you provide a minimal reproduction example?

`react-native-edge-to-edge` is not using deprecated API so it should work fine. If you can provide a minimal reproduction example, then it can definitely speed up the process of fixing the problem 👀

**Comment by rranjan14** [comment]
@kirillzyusko unfortunately I do not see this happening in a new project. I suspect this is a configuration issue at my end. I have tried too many things to solve but the outcome is always same. Also, what i observed is, when the value of top inset returned is not 0 from `react-native-safe-area-context` after first render, the white space is gone. What's happening is, after first render, the values returned from `useSafeAreaInsets` hook is 0 for all the sides.

**Comment by kirillzyusko** [comment]
@rranjan14 but you can try to simplify your project. In the end you'll end up with a project, where you have 5-6 files and where the issue is reproducible. And then you can share that project (or create a new project and copy all changes from your simplified project to a new project) 👀

**Comment by rranjan14** [comment]
@kirillzyusko its a big task. I'd surely try this. Will share the simplified project when I can. Thanks a lot! 🫡

**Comment by rotoxl** [comment]
In case it helps, I had the same problem and adding `statusBarTranslucent` and `navigationBarTranslucent` to the provider fixed it

```
  <KeyboardProvider statusBarTranslucent navigationBarTranslucent>
   {children}
</KeyboardProvider>
```

**Comment by Tobbe** [comment]
> ![image](https://github.com/user-attachments/assets/64a07352-bd28-47a0-a4c9-8ab0b35082d4)
> I am also facing this issue on android. I am getting a white space at the top of the screen. The space is roughly equal to the height of the statusbar. This is happening with `react-native-screens >=3.31.0`

I've been seeing this issue as well. But only sometimes. And I have not been able to reproduce it reliably. 

Sometimes I also have the same thing happening but at the bottom of the screen when the keyboard is visible. Like this (notice the yellow bar above the keyboard that hides part of the button)

![image](https://github.com/user-attachments/assets/95532b47-8796-4981-a78d-25a851477c49)

So far uninstalling and then reinstalling the app has been the only way to fix it for me once it starts doing that. Here it works as expected. (No extra yellow bar between the top of the keyboard and what's visible on the screen)
![image](https://github.com/user-attachments/assets/71be38a0-3ab0-4a7c-883c-d90f549306f0)

**Comment by Tobbe** [comment]
Working on a reproduction I see weird things happening both at the top and the bottom of the screen at the same time
![image](https://github.com/user-attachments/assets/77be9f61-2268-4c5b-8445-c01e0f5f9adb)

Here's a reproduction
https://github.com/Tobbe/rnkc-keyboard-padding

**Comment by rranjan14** [comment]
> Working on a reproduction I see weird things happening both at the top and the bottom of the screen at the same time ![image](https://private-user-images.githubusercontent.com/30793/397749839-7bdcc822-b57f-4c26-9143-b99f536e0573.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDE5MTAsIm5iZiI6MTczNDcwMTYxMCwicGF0aCI6Ii8zMDc5My8zOTc3NDk4MzktN2JkY2M4MjItYjU3Zi00YzI2LTkxNDMtYjk5ZjUzNmUwNTczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDEzMzMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE0NTIzNjI5ZjM1MTRhZTI5NTVkZWM3NTFhNDJmMWY3ZTUyMTRlYTk2ODM3MGIyODZjMDZhOGUyNTIyOWFiOTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.DIdogEDzUCgTdzFif-t_nQNdr-4t0xu8duokyv5ZJ9Y)
> 
> Here's a link to the code used for the screenshot above https://github.com/Tobbe/rnkc-keyboard-padding
> 
> If I add `enabled={false}` to `<KeyboardController>` and reload the app (by pressing `r`) I see this, which doesn't have the extra padding neither at the top nor the bottom ![image](https://private-user-images.githubusercontent.com/30793/397750243-ed50520f-6d5d-4d82-bd54-88ed34f448fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDE5MTAsIm5iZiI6MTczNDcwMTYxMCwicGF0aCI6Ii8zMDc5My8zOTc3NTAyNDMtZWQ1MDUyMGYtNmQ1ZC00ZDgyLWJkNTQtODhlZDM0ZjQ0OGZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDEzMzMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkZWE0MTU2MmUwMzU4NjM1MzczYjgzMDA4NDNmYzY4ZjUwNTNlODhkNjY0YmIxNTc2MWQ3MDA2ODEzNDZhMjEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.0iFsS9hSrPz6p_Ygc-p-_lckZ-Ta0uxNlePRYhD_9nM)
> 
> I simplified the code for the reproduction even more. Now it's just two files.
> 
> ![image](https://private-user-images.githubusercontent.com/30793/397753941-2fdda2fa-7a4b-4d91-b557-4628a125c573.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ3MDE5MTAsIm5iZiI6MTczNDcwMTYxMCwicGF0aCI6Ii8zMDc5My8zOTc3NTM5NDEtMmZkZGEyZmEtN2E0Yi00ZDkxLWI1NTctNDYyOGExMjVjNTczLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjIwVDEzMzMzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFkMThkZWJkMTgyNWFmZmVjNWQzNDYzZGVkYjE4YzY2MjY3MjJjZjk0ZDRhMGQzMjVjNjBkYzZiOGY3MGQ5NDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.BVCeBjNtnXyWuZ12N4-OmiKZsnYzffzKCBhO85-ODXw)
> 
> ```tsx
> // _layout.tsx
> 
> import { Stack } from "expo-router";
> import { KeyboardProvider } from "react-native-keyboard-controller";
> 
> export default function RootLayout() {
>   return (
>     <KeyboardProvider>
>       <Stack />
>     </KeyboardProvider>
>   );
> }
> ```
> 
> ```tsx
> // index.tsx
> 
> import React from "react";
> import { Text, ScrollView, StyleSheet } from "react-native";
> 
> export default function App() {
>   return (
>     <ScrollView contentContainerStyle={styles.scrollViewContainer}>
>       <Text>Foobar</Text>
>     </ScrollView>
>   );
> }
> 
> const styles = StyleSheet.create({
>   scrollViewContainer: {
>     backgroundColor: "lightpink",
>     flexGrow: 1,
>   },
> });
> ```

Can you try adding the `react-native-edge-to-edge` package and check with `enabled` prop set as true? It might solve it for you.

**Comment by Tobbe** [comment]
@rranjan14 Thanks! That seems to have improved things 🙂 Now I just need to start slowly adding things back one by one and see how far I can take it until my app breaks again 😆

**Comment by kirillzyusko** [comment]
`react-native-screens@4` also might be not always compatible with `keyboard-controller`: https://github.com/software-mansion/react-native-screens/issues/2554 😔

**Comment by Tobbe** [comment]
@kirillzyusko Thank you! That might explain why rnkc sometimes just stops working altogether for me. Is there any kind of workaround right now? Or do I need to downgrade to earlier version of react-native-screens?

**Comment by kirillzyusko** [comment]
> Or do I need to downgrade to earlier version of react-native-screens?

Well, this is the most proper way to resolve that now 👍 

Another workaround that I discovered is mounting `KeyboardProvider` with some delay (however it's far from being perfect approach). I'm going to look into why the integration with RNS is broken and how it can be fixed 👀

**Comment by kirillzyusko** [comment]
@Tobbe would you mind to check if https://github.com/kirillzyusko/react-native-keyboard-controller/pull/744 fixes a problem for you or not? 👀 

It may fix a problem 👀

**Comment by kirillzyusko** [comment]
@e-biba-ofti I also believe that https://github.com/kirillzyusko/react-native-keyboard-controller/pull/744 should fix the reported problem (after re-reading our conversation). Could you please check that? 🙏

**Comment by kirillzyusko** [comment]
The issue was automatically closed because I merged the PR. I'm going to publish a fix to npm under `1.15.2` tag.

If you still encounter this problem, then please open a new issue with all necessary information (such as reproduction example etc.)

I strongly believe, that the merged PR should resolve this problem! ❤️

**Comment by RayKay91** [comment]
Can confirm it fixed the issue for me! I was using react-native-edge-to-edge as a workaround but just removed that and bumped the keyboard-controller to 1.15.2 and there's no extra padding above or below. Thank you! :D

**Comment by kirillzyusko** [comment]
Yooohooo @RayKay91 🎉 Glad it resolved the problem! 😊

**Comment by Tobbe** [comment]
@kirillzyusko Thanks for working on this issue! 
I'm away spending some time with family over the holidays. I'll try the new version when I'm back at work again.

**Comment by kirillzyusko** [comment]
> I'm away spending some time with family over the holidays. I'll try the new version when I'm back at work again.

Sure ❤️

**Comment by RahulMandyal1** [comment]
Hi @kirillzyusko 
I have been using:
"react-native-screens": "~4.4.0",
"react-native-keyboard-controller": "^1.17.1"

Since upgrading to Expo SDK 52, I'm noticing extra space at the bottom of the screen.
Do I need to downgrade any of these libraries to fix the issue, or is there a recommended solution?

**Comment by agusstingalvan** [comment]
> Hi [@kirillzyusko](https://github.com/kirillzyusko) I have been using: "react-native-screens": "~4.4.0", "react-native-keyboard-controller": "^1.17.1"
> 
> Since upgrading to Expo SDK 52, I'm noticing extra space at the bottom of the screen. Do I need to downgrade any of these libraries to fix the issue, or is there a recommended solution?

Hi! I have same issue and same versions

**Comment by JapjyotSuri** [comment]
After adding expo updates config to my expo app with expo version as 52.0.17 I am getting the white padding issue sometimes on reloading the apk made using eas build but when I remove the expo updates config the issue doesn't happen and after  specifing statusBarTranslucent and navigationBarTranslucent sometimes on reloading the ui is going under the status bar and navigation bar can you please help me solve this issue?
The version for react-native-keyboard-controller and react-native-screens are given below:
"react-native-screens": "~4.1.0",
"react-native-keyboard-controller": "^1.16.6",

**Comment by kirillzyusko** [comment]
@JapjyotSuri can you try to update react-native-screens to 4.11+ as was recommended a few posts above?

Additionally - can I reproduce the problem without expo-updates (i.e. by reloading the app)? If no, how can I reproduce the problem locally?..

**Comment by JapjyotSuri** [comment]
@kirillzyusko I have upgraded to the latest version of react-native-screens that is 4.13.1 and I am getting the same issue.The problem is not happening without expo updates config and there is no way to test it in on local build I am also testing the app by making a eas build and sending ota updates The config for expo updates that I have added is given below:

`updates: {
      checkAutomatically: 'NEVER',
      url: 'my project url',
    },
    runtimeVersion: {
      policy: 'appVersion',
    },`

and I am checking if an update is available manually by adding the below condition in my root index file

`if (update.isAvailable)`

The same issue is coming when I set  checkAutomatically to ON_LOAD in the app.config.js

**Comment by kirillzyusko** [comment]
@JapjyotSuri I published `1.18.1` which _potentially_ may have a fix for this problem. Would you mind to try it out?

If it doesn't fix the problem, then please open a new issue 🙏 

If you could add a simple reproduction example + instructions how can I reproduce it then I'll highly appreciate it and it can significantly reduce time for fixing the issue.
I've never worked with `expo-updates`, so detailed instructions what to press etc. are highly desirable! ❤️ 

P. S. after re-reading the messages above I don't think `1.18.1` will fix it 😔 So don't hesitate to open a new issue 🙌 I can also share my discord nickname so that we can have a chat there 👀

**Comment by kirillzyusko** [comment]
@JapjyotSuri oh, I think we have open issue in https://github.com/kirillzyusko/react-native-keyboard-controller/issues/993

Feel free to add more details 🙌
